docs/core/deploying/native-aot/diagnostics.md

```diff
@@ -0,0 +1,54 @@
+---
+title: AOT Diagnostics
+description: Learn about diagnostics in NativeAOT applications
+author: lakshanf
+ms.author: lakshanf
+ms.date: 08/07/2023
+---
+# Diagnostics in NativeAOT applications
+
+The long term goal for NativeAOT application diagnostics is to provide the rich diagnostic experience that developers expect out of a .NET application. .NET diagnostic experience was built over multiple releases and getting customer feedback on diagnostic was a critical part of that journey. NativeAOT wants to follow a similar path where the right diagnostic experience is built from prioritized customer feedback in multiple releases.
+
+NativeAOT applications have the following characteristics that are important to keep in mind in diagnostic scenarios:
+
+- Trimming as a first-class feature in NativeAOT application. Unused code in a NativeAOT application is stripped out of the final binary. Any unbounded reflection used by the application including its dependent libraries will be trimmed. NativeAOT compiler will generate warnings in such cases, and it is critical that these warnings are fixed. For example, suppressing any such warnings without careful analysis can lead to hard to debug production failures.
```

Comment:
```suggestion
- Trimming as a first-class feature in NativeAOT application. Unused code in a NativeAOT application is stripped out of the final binary. Any unbounded reflection used by the application, including its dependent libraries, is trimmed. The NativeAOT compiler will generate warnings in such cases, and it's critical that these warnings are fixed. For example, if you suppress any warnings without careful analysis, it can lead to hard to debug production failures.
```